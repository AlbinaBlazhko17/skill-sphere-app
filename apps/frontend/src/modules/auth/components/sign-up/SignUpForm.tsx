import {
	Button,
	Form,
	FormControl,
	FormField,
	FormItem,
	FormLabel,
	FormMessage,
	Input,
} from '@/components/ui';
import { useSignUpForm } from './use-sign-up-form';

export const SignUpForm = () => {
	const { form, onSubmit } = useSignUpForm();

	return (
		<Form {...form}>
			<form onSubmit={onSubmit} className={'space-y-6 px-2'}>
				<FormField
					name={'firstName'}
					control={form.control}
					render={({ field }) => (
						<FormItem>
							<FormLabel htmlFor={'firstName'}>First name</FormLabel>
							<FormControl>
								<Input {...field} id={'firstName'} />
							</FormControl>
							<FormMessage />
						</FormItem>
					)}
				/>

				<FormField
					name={'lastName'}
					control={form.control}
					render={({ field }) => (
						<FormItem>
							<FormLabel htmlFor={'lastName'}>Last name</FormLabel>
							<FormControl>
								<Input {...field} id={'lastName'} />
							</FormControl>
							<FormMessage />
						</FormItem>
					)}
				/>

				<FormField
					name={'email'}
					control={form.control}
					render={({ field }) => (
						<FormItem>
							<FormLabel htmlFor={'email'}>Email</FormLabel>
							<FormControl>
								<Input {...field} id={'email'} />
							</FormControl>
							<FormMessage />
						</FormItem>
					)}
				/>

				<FormField
					name={'password'}
					control={form.control}
					render={({ field }) => (
						<FormItem>
							<FormLabel htmlFor={'password'}>Password</FormLabel>
							<FormControl>
								<Input {...field} id={'password'} type={'password'} />
							</FormControl>
							<FormMessage />
						</FormItem>
					)}
				/>

				<div className={'flex justify-end'}>
					<Button
						type={'submit'}
						className={'h-10 w-32'}
						isLoading={form.formState.isSubmitting}
					>
						Sign up
					</Button>
				</div>
			</form>
		</Form>
	);
};
