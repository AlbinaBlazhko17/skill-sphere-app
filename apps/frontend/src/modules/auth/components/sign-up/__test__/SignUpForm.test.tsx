import { SignUpForm } from '../SignUpForm';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('SignUpForm', () => {
  it('should render the form', () => {
    const wrapper = render(<SignUpForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the first name field', () => {
    const wrapper = render(<SignUpForm />);
    expect(wrapper.getByLabelText('First name')).toBeInTheDocument();
  });
});
