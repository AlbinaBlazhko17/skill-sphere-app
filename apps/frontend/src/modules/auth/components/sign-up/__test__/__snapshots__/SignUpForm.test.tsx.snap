// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="space-y-6 px-2"
      >
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="firstName"
          >
            First name
          </label>
          <input
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="firstName"
            name="firstName"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="lastName"
          >
            Last name
          </label>
          <input
            aria-describedby=":r1:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="lastName"
            name="lastName"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="email"
          >
            Email
          </label>
          <input
            aria-describedby=":r2:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="email"
            name="email"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="password"
          >
            Password
          </label>
          <input
            aria-describedby=":r3:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="flex justify-end"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-cyan-400 text-violet-950 shadow-xs hover:bg-cyan-400 px-4 py-2 has-[>svg]:px-3 w-32 h-10"
            data-slot="button"
            type="submit"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="space-y-6 px-2"
    >
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="firstName"
        >
          First name
        </label>
        <input
          aria-describedby=":r0:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="firstName"
          name="firstName"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="lastName"
        >
          Last name
        </label>
        <input
          aria-describedby=":r1:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="lastName"
          name="lastName"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="email"
        >
          Email
        </label>
        <input
          aria-describedby=":r2:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="email"
          name="email"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="password"
        >
          Password
        </label>
        <input
          aria-describedby=":r3:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-cyan-400 text-violet-950 shadow-xs hover:bg-cyan-400 px-4 py-2 has-[>svg]:px-3 w-32 h-10"
          data-slot="button"
          type="submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SignUpForm should render the form 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="space-y-6 px-2"
      >
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="firstName"
          >
            First name
          </label>
          <input
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="firstName"
            name="firstName"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="lastName"
          >
            Last name
          </label>
          <input
            aria-describedby=":r1:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="lastName"
            name="lastName"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="email"
          >
            Email
          </label>
          <input
            aria-describedby=":r2:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="email"
            name="email"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
            data-error="false"
            data-slot="form-label"
            for="password"
          >
            Password
          </label>
          <input
            aria-describedby=":r3:-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="flex justify-end"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-cyan-400 text-violet-950 shadow-xs hover:bg-cyan-400 px-4 py-2 has-[>svg]:px-3 w-32 h-10"
            data-slot="button"
            type="submit"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="space-y-6 px-2"
    >
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="firstName"
        >
          First name
        </label>
        <input
          aria-describedby=":r0:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="firstName"
          name="firstName"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="lastName"
        >
          Last name
        </label>
        <input
          aria-describedby=":r1:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="lastName"
          name="lastName"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="email"
        >
          Email
        </label>
        <input
          aria-describedby=":r2:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="email"
          name="email"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="text-sm text-violet-50 leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive-foreground"
          data-error="false"
          data-slot="form-label"
          for="password"
        >
          Password
        </label>
        <input
          aria-describedby=":r3:-form-item-description"
          aria-invalid="false"
          class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-cyan-400 text-violet-950 shadow-xs hover:bg-cyan-400 px-4 py-2 has-[>svg]:px-3 w-32 h-10"
          data-slot="button"
          type="submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
